xdsl/dialects/irdl.py

```diff
@@ -170,6 +176,20 @@ def verify_(self) -> None:
         if not isinstance(self.attributes["name"], StringAttr):
             raise ValueError("name attribute must be a string attribute")
 
+    def get_operands(self) -> OperandsOp | None:
```

Comment:
These accessors at most return one op, right? I am guessing that operations can have multiple operands in IRDL, right? Or is this then just modeled differently?

Overall, I am quite surprised that these look so different from the other accessors, as it feels like they do the same thing (loop over a region, find all things that have a specific type).

Reply:
So in IRDL (the way it's currently modeled in MLIR), operations have at most a single `OperandsOp`, and the `OperandsOp` contains a list of attributes, which represent the different operands.

Reply:
Okay, that's what I expected :+1:

Reply:
Though it seems that the `mul` op in https://github.com/xdslproject/xdsl/blob/main/tests/filecheck/irdl-printer/cmath_program.xdsl does not follow that principle. I guess strengthen the verifier might make sense at some point.

Reply:
Hmm good point, I'll fix that in a separate PR, and also remove that printer (since we can use the MLIR printer now)!